models/MonoWave.py

```diff
@@ -1,24 +1,60 @@
 from __future__ import annotations
+
 import numpy as np
+
 from models.functions import hi, lo, next_hi, next_lo
 
+
 class MonoWave:
+    """
+    A class representing a monowave pattern in financial time series data.
+
+    Attributes:
+    - lows_arr (numpy.array): An array containing the lows of the monowave pattern.
+    - highs_arr (numpy.array): An array containing the highs of the monowave pattern.
+    - dates_arr (numpy.array): An array containing the dates corresponding to the monowave pattern.
+    - idx_start (int): The index of the starting point of the monowave pattern.
+    - skip_n (int, optional): The number of elements to skip during pattern analysis (default is 0).
+    - idx_end (int): The index of the ending point of the monowave pattern.
+    - count (int): The count of the monowave (e.g., 1, 2, A, B, etc.).
+    - degree (int): The degree of the monowave (1 = lowest timeframe level, 2 as soon as a pattern like 12345 is found,
+    etc.).
+    - date_start (str): The start date of the monowave pattern.
+    - date_end (str): The end date of the monowave pattern.
+    - low (float): The lowest value in the monowave pattern.
+    - high (float): The highest value in the monowave pattern.
+    - low_idx (int): The index of the lowest value in the monowave pattern.
+    - high_idx (int): The index of the highest value in the monowave pattern.
+
+    Properties:
+    - labels (str): A property that returns the count of the monowave as a string.
+    - length (float): A property that returns the absolute difference between high and low values.
+    - duration (int): A property that returns the duration (index difference) of the monowave.
+
+    Methods:
+    - from_wavepattern(cls, wave_pattern): A class method to create a MonoWave object from a given wave pattern.
+
+    Note:
+    - This class is designed for analyzing financial time series data and identifying monowave patterns.
+    """
+
     def __init__(self,
                  lows: np.array,
                  highs: np.array,
                  dates: np.array,
                  idx_start: int,
                  skip: int = 0):
 
+        # Constructor initializes the attributes
         self.lows_arr = lows
         self.highs_arr = highs
         self.dates_arr = dates
         self.skip_n = skip
         self.idx_start = idx_start
         self.idx_end = int
 
-        self.count = int  # the count of the monowave, e.g. 1, 2, A, B, etc
-        self.degree = 1  # 1 = lowest timeframe level, 2 as soon as a e.g. 12345 is found etc.
+        self.count = int
```

Comment:
should be more like `.count = 0 `or `.count: int` as type  hint not assigned to class int (but i know that its old code anyway and i m overdue to clean it up myself :-) )